test/exoscale/compute/api/http_test.clj

```diff
@@ -109,3 +111,104 @@
+                                                                     {:jobid job-id}}})
+                           :queryAsyncJobResult (constantly {:status 200
+                                                             :body job-resp})}
+      (is (= job-result
+             (auspex/unwrap (http/request!! {:endpoint "http://localhost:8080"
+                                             :api-key "foo"
+                                             :api-secret "bar"}
+                                            "deleteSnapshot"
+                                            {:id (rand-uuid)}))
+             (auspex/unwrap (http/request!! {:endpoint "http://localhost:8080"
+                                             :api-key "foo"
+                                             :api-secret "bar"
+                                             :throw-on-job-failure? true}
+                                            "deleteSnapshot"
+                                            {:id (rand-uuid)})))))))
+(deftest listing-commands
+  (let [template-id (rand-uuid)
+        error       (format "Unable to execute API command listtemplates due to invalid value. entity %s does not exist." template-id)
+        error-resp  {:listtemplatesresponse
+                     {:cserrorcode 9999
+                      :errorcode 431
+                      :errortext error}}]
+    (utils/with-http 8080 {:listVirtualMachines  (constantly {:status 200
+                                                              :body {:listvirtualmachinesresponse
+                                                                     {:count 0
+                                                                      :virtualmachine []}}})
+
+                           :listTemplates (constantly {:status 431
+                                                       :body error-resp})}
+      (is (= []
+             (auspex/unwrap (http/request!! {:endpoint "http://localhost:8080"
+                                             :api-key "foo"
+                                             :api-secret "bar"}
+                                            "listVirtualMachines"
+                                            {}))))
+      (try
+        (auspex/unwrap (http/request!! {:endpoint "http://localhost:8080"
+                                        :api-key "foo"
+                                        :api-secret "bar"}
+                                       "listTemplates"
+                                       {:id template-id}))
+        (is false "call should have failed")
+        (catch Exception e
+          (clojure.pprint/pprint e)
+          (let [{:keys [response]} (ex-data e)]
+            (is (= 431 (:status response))))
+          (is (= "HTTP Error"
+                 (ex-message e)))
+          #_(let [{:keys [body status]} (ex-data e)]
+              (is (= status 431)
+                  (= (json/parse-string body)
+                     error-resp))))))))
```

Comment:
In Aleph version, in case of error we used to return a map containing _status_ and _body_ (as a string)
We are now returning a map with a _response_ key that contains some information about the failure (status, body)
It seems that response body is not serialized into a string/map. We have to retrieve it from `jdk.internal.net.http.ResponseSubscribers/HttpResponseInputStream`
@mpenet is it expected/desired ?